### DISCUSSION TITLE
How to export camera animation to video or image sequence?

### DISCUSSION THREAD

**Discussion by NcMasters** [NVlabs/instant-ngp#191] [Q&A] [answered]
/\

**Comment by Tom94** [comment, marked as answer]
Hi there, I think this answer will help you: https://github.com/NVlabs/instant-ngp/issues/31

### Pasted from there

The Python bindings allow this. Here's a relevant excerpt from our internal scripts. (We're working on pushing more of them public in the future.)

```
def render_video(resolution, numframes, spp, fps, exposure=0):
	for i in tqdm(list(range(min(numframes, last_frame+1))), unit="frames", desc=f"Rendering"):
		testbed.camera_smoothing = i > 0
		frame = testbed.render(resolution[0], resolution[1], spp, True, float(i)/numframes, float(i + 1)/numframes, fps, shutter_fraction=0.5)
		common.write_image(f"{i:04d}.jpg", np.clip(frame * 2**exposure, 0.0, 1.0), quality=100)

testbed.load_camera_path(<path to camera_path.json>)

n_seconds = 5
fps = 60
render_video([1920, 1080], int(n_seconds/fps), spp=8, fps=fps)
```

Note: the `testbed.camera_smoothing` property enables an additional exponential averaging of camera matrices over time (log-space) on top of the spline-based camera path.

**Reply by NcMasters** [reply]
Tysm for the answer but how do i run the code thing? I am new to this

**Reply by bycloudai** [reply]
I wrote a temporary solution using the code the author gave. Hopefully that helps.

https://github.com/bycloudai/instant-ngp-Windows#rendering-custom-camera-path

I kind of don't get how the `n_seconds` and `fps` relates to the camera/video speed, so please enlighten me and excuse me for my messy code.

**Reply by DuckyBlender** [reply]
How and where do I run this?

**Reply by thoi86** [reply]
Hi! I tried installing your fork @bycloudai and everything works, except video save. The render.py script does not end up in the instant-ngp folder. This causes the error:  error: python: can't open file 'C:\Users\...\Desktop\instant-ngp-Windows-master\instant-ngp\scripts\render.py': [Errno 2] No such file or directory

What steps am I missing in order to be able to save video? I have tried installing it with your guide serval times. 

Best
